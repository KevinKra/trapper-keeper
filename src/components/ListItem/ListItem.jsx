import React from "react";
import PropTypes from "prop-types";
import "./ListItem.scss";

const ListItem = props => {
  const { userTask, complete } = props;
  if (complete) {
    return (
      <li className="ListItem completedTask">
        <i className="material-icons checked">check</i>
        <p>{userTask}</p>
      </li>
    );
  } else {
    return (
      <li className="ListItem activeTask">
        <i className="material-icons">crop_square</i>
        <p>{userTask}</p>
      </li>
    );
  }
};

ListItem.propTypes = {
  complete: PropTypes.bool,
  id: PropTypes.number,
  userTask: PropTypes.string
};

export default ListItem;
