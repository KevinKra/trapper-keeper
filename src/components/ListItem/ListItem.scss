@import "../../variables.scss";

.ListItem {
  display: flex;
  align-items: flex-start;
  word-wrap: break-word;
  text-align: left;
  i {
    @include list-icon-styles;
    margin: 5px 5px 0 0;
  }
  .checked {
    font-size: 20px;
    padding-left: 2px;
  }
  p {
    margin: 3px 0;
  }
}

.activeTask {
  color: $darkFont;
}

.completedTask p {
  text-decoration: line-through;
  color: $lightgrey;
}
