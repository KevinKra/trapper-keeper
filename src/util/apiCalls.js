export const getAllLists = async () => {
  const res = await fetch(`${process.env.REACT_APP_BASE_URL}/api/v1/lists`);
  if (!res.ok) {
    throw new Error("Could not get lists");
  }
  return await res.json();
};

export const addNewList = async list => {
  const res = await fetch(`${process.env.REACT_APP_BASE_URL}/api/v1/lists`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(list)
  });
  if (!res.ok) {
    throw new Error("Could not save new list");
  }
  return await res.json();
};

export const getListData = async id => {
  const res = await fetch(
    `${process.env.REACT_APP_BASE_URL}/api/v1/lists/${id}`
  );
  if (!res.ok) {
    throw new Error("Could not fetch list with that id");
  }
  return await res.json();
};

export const deleteList = async id => {
  const res = await fetch(
    `${process.env.REACT_APP_BASE_URL}/api/v1/lists/${id}`,
    {
      method: "DELETE"
    }
  );
  if (!res.ok) {
    throw new Error("Could not delete card");
  }
  return await res.json();
};
