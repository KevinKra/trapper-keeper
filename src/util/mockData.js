export const mockNote = {
	id: Date.now(),
	userTask: 'Hello, world!',
	complete: false
};

export const mockList = {
	title: 'My List',
	notes: [mockNote, mockNote, mockNote]
};

export const mockLists = [mockList, mockList, mockList];
