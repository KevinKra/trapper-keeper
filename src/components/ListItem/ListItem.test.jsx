import React from 'react';
import ListItem from './ListItem';
import * as mockData from '../../util/mockData';
import { shallow } from 'enzyme';

describe('ListItem', () => {
	it('should match the snapshot', () => {
		const wrapper = shallow(<ListItem {...mockData.mockNote} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should match the snapshot if not complete', () => {
		const wrapper = shallow(<ListItem {...mockData.mockNote} complete={true} />);
		expect(wrapper).toMatchSnapshot();
	});
});
