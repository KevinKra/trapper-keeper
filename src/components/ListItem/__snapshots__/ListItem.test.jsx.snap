// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`ListItem should match the snapshot if not complete 1`] = `ShallowWrapper {}`;
